### ISSUE TITLE
Kevin's User Directory v2 / north star ticket

### ISSUE THREAD

**Issue by owocki** [gitcoinco/web#7606]
Hey all, 

IMHO user directory is a huge strategic win for us if we can get it right. This ticket is designed to articulate the why/what/how behind that. - I figured I'd spend a moment articulating my view on where the user directory could go and why it's so important to invest resources there.  

This ticket is meant to be a north star for the user directory work, an ultimate destination of where it could go if we execute well.

To state it explicitly - this ticket is not meant to jeer prioritizes in a new direction.  These 'north star' thoughts are secondary to the ongoing active tickets ( https://github.com/gitcoinco/web/issues/7334 https://github.com/gitcoinco/web/issues/7579 ) and to connor/justins feedback on how they could operationalize the user directory as it exists today - which are more urgent than the "where is the ultimate destination" stuff.  

## where we are / why it matters

First; why is the user directory SO important to me?  Well, Gitcoin is a double sided market that connects coders to funders.  The thing about this double sided market is that our funders want THE RIGHT coder for them, not just any coder.  The inventory is non-fungible!  Some examples of this:

1.   When Scott sells a partnership, it often comes along with a question "Well how many users do we have that fit criteria X, and how do we activate those users in a targeted compelling way?" comes up..  
2. When someone contacts me on twitter DMs being like "hey im looking for a coder who fits X profile", I  really want to show them how many coders we have on the platform that fit that profile, and the natural progression of that conversation is to allow them to market to our userbase (possibly for a fee).
3. There are numerous times in which someone launches a grant/hackathon/partnership, in which we could do a better job of promoting it if we had the ability to micro-target an audience on the site. 
4. People use gitcoin for our network of coders. We've even gotten feedback that they "put up with the product to get the coders".  The UD could allow them to get direct, filtered, unfettered, access to our network of coders.

TLDR - The user directory is SUPER important because, eventually, itll allow us to do analytics on our userbase + activate those users (or if we dont have the inventory, tell Scott we cant do the partnership and/or launch our own marketing campaign to beef up our presence to a specific audience) in a very quick, easy, effective, way.  Itll allow us to traverse the entire coder engagement flow end to end:

```
source coder > select the right coder for me > activate coder
```

## where we are going

Right now the user directory 'kinda' works for this use case in a clunky way.    Right now I have to de-dupe 'python' and 'Python' , some of the filters dont work, i can't deselect a user, (all of these are tracked in  ( https://github.com/gitcoinco/web/issues/7334 https://github.com/gitcoinco/web/issues/7579 )  ) and the export to CSV requires me to load the data into another CRM in order to contact the users.

In the future, I'm envisioning that the user directory can not only allow you to query users, but also to measure audience sizes, share user queries, and message users directly on the platform.  Perhaps we will even be able to bulk-tip users as we message them, allowing us to pay users for their time/attention in reading messages from the user directory (similar to the earn.com use case that propelled them to a giant exit).

Why am I excited about bulk messaging/tipping?

My hunch is informed by the following:

1. The 'message all my funders' functionality on grants was a huge success.  It turned grants from a funding platform to a funding platform + marketing platform.
2. Grants went through a stepwise increase in popularity when we went from 1:1 funding of grants to bulk funding grants
3. Tipping is one of the simplest features on Gitcoin, but also one of the most popular.  It's driven $1mm in GMV so far.
4. Earn.com proved that people are willing to be marketed to if they are paid for it.

## goal state

IMHO user directory is a huge strategic win for us if we can get it right. 

We are a business that connects coders to the people who want to fund them. The goal state for this initaitive is that we can very QUICKLY and EFFECTIVELY traverse the following funnel:

```
source coder > select the right coder for me > activate coder
```

The user directory, in its final state, allows us to filter/measure/engage our audience super effectively.

We can/should iterate towards this goal state through a series of successive releases that makes doing this quick, easy, and maybe even enjoyable . :)

**Comment by frankchen07** [comment]
Everything makes sense to me here. The way you have it outlined here, the UD is a major product, and its use hinges around the actual "search", instead of us abstracting it away. That was probably the primary use case difference and why I had a different opinion about it.

My hesitation from before was mainly that our value proposition should take care of all the matching necessary for the sponsor (so they don't have to specifically look for anyone). 

The intuition about UD outline above seem like great hypotheses to test. Having the user directory so we can use it when we get asked marketing questions is part of that, but really, having it ready so that sponsors can quickly traverse "source > select > activate" is the true test. At the very least, it eventually removes me from the equation (less querying for me).

I agree that we're still in the stages of getting a better working prototype up, and part of that is a data-cleaning exercise. There's some action steps from the data end that would help clean things up and I think there can be more done with de-duping tags and making sure we get the cleanest data possible.

1. de-dupe github tabs
2. deprecate old onboarding and the ability to add custom tags
3. add new onboarding information to user directory
4. nail down the columns/metrics we for sure want filters by (categorize into product metrics / interests / skills, etc)

**Comment by owocki** [comment]
pasting what i said to alisa this AM on DMs about 'not swallowing the elephant' by trying to to have it work 'automatically' at first, instead just getting it to a level that we can use it interanlly first, then iterating towards external usage.

```

thanks for the thoughtful feedback.

two things we can do to meet our customer expectations RE: bountys ‘automatically come’.
1. we can adjust our expectations to let them know that they should promote their bounty (this could be good for our viral loops)
2. for an upsell, we can do some sort of gitcoin boost- where we mechanical turk the invites to the bounty behind the scenes. wed of course need a working UD for this

i think there are a bunch of things we could work backwards from if we could easily, effectively, query/engage our userbase.    we shoudl always start with

1. “gitcoin can effectively do this manually”
then move on to
2. then move on to “gitcoin can effectively do this automatically”

i get a little antsy + worry were trying to swallow an elephant when i hear things like “how do we automatically... X...” from the start bc i think we might be swalling an elephant. need to solve it manually so that our team can do it first; then move on to doing it  automatically.

> You kind of expect uber drivers to just be there and ready to take you I think.
one diff between us and the uber use case is that theres way more criteria for a successfulmatch for us than uber:

uber: safe, secure, quick
gitcoin: reliable, same language, same timezone or good at async, can communicate well, can code in X well, can iterate on feedback well. i just think software development is more abstract than driving, and coders are non-fungible inventory as opposed to drivers who are fungible, so its going to be a stepwise harder thing to solve for.. not that we shouldnt shoot high, just kind of articulating the problem (and why its so important to have data + indexed users to solve it)

```
